src/pfe/file-watcher/server/test/functional-test/suites/generic/index.ts

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import mocha from "mocha";
+import { SocketIO } from "../../lib/socket-io";
+
+import LocaleTest from "./tests/locale";
+import LoggingTest from "./tests/logging";
+import ProjectTest from "./tests/project";
+import WorkspaceTest from "./tests/workspace";
+
+/**
+ * @class
+ * @description The generic test suite for PFE Turbine Module.
+ *
+ */
+export default class GenericTestSuite {
+    suite: string;
+    socket: SocketIO;
+
+    localeTest: LocaleTest;
+    loggingTest: LoggingTest;
+    projectTest: ProjectTest;
+    workspaceTest: WorkspaceTest;
+
+    /**
+     * @constructor
+     * @description The constructor of the generic test suite.
+     *
+     */
+    constructor(socket: SocketIO) {
+        this.suite = "Generic Suite";
+        this.socket = socket;
+
+        this.localeTest = new LocaleTest("locale test");
+        this.loggingTest = new LoggingTest("logging test");
+        this.projectTest = new ProjectTest("project test");
+        this.workspaceTest = new WorkspaceTest("workspace test");
+    }
+
+    /**
+     * @function
+     * @description The run test function that triggers all generic test classes.
+     *
+     */
+    runTest(runOnly?: boolean): void {
```

Comment:
I think this parameter `runOnly` is used for only running individual test suite? If so does that mean if we want to run individual test suite we have to manually change the test case to make `runOnly` true then copy the test case to PFE container?

Generally for running individual test suite/unit, Mocha has option
`-g, --grep <pattern>            only run tests matching <pattern>` but you need to have nested `describe` to describe the test suite then Mocha can grep the specific test suite/unit

Reply:
Nope. You don't have to manually copy anything to the container. Only set `runOnly` to true and run `./test.sh local` and it will only run that test suite.

Reply:
But all the test cases we run are inside container, does the `runOnly` you set is against you local test case? If you directly run `./test.sh`, does it still trigger the old test cases inside container?

Reply:
`runOnly` set is applied to the test cases inside the container because the `./test.sh` script copies over the `/test` directory to the container before running the test.

Reply:
Verified, it works, nice!